adm_config_set.php

```diff
@@ -102,177 +102,32 @@
 	$t_type = $f_type;
 }
 
-switch( $t_type ) {
-	case CONFIG_TYPE_STRING:
-		$t_value = $f_value;
-		break;
-	case CONFIG_TYPE_INT:
-		$t_value = (integer)constant_replace( trim( $f_value ) );
-		break;
-	case CONFIG_TYPE_FLOAT:
-		$t_value = (float)constant_replace( trim( $f_value ) );
-		break;
-	case CONFIG_TYPE_COMPLEX:
-	default:
-		$t_value = process_complex_value( $f_value );
-		break;
+# Parse the value
+if( $t_type == CONFIG_TYPE_STRING ) {
+	# Return strings as is
+	$t_value = $f_value;
+} else {
+	try {
+		$t_parser = new ConfigParser( $f_value );
+		$t_value = $t_parser->parse( ConfigParser::EXTRA_TOKENS_IGNORE );
```

Comment:
Does this code need to path in this parameter?  Or should it be the default for the parser?